src/v/kafka/server/handlers/create_topics.cc

```diff
@@ -171,6 +171,37 @@ ss::future<response_ptr> create_topics_handler::handle(
         valid_range_end = unauthorized_it;
     }
 
+    const auto additional_resources_func = [&request, has_cluster_auth]() {
+        std::vector<model::topic> topics;
+        if (!has_cluster_auth) {
+            return topics;
+        }
+        topics.reserve(request.data.topics.size());
+        std::transform(
+          request.data.topics.begin(),
+          request.data.topics.end(),
+          std::back_inserter(topics),
+          [](const creatable_topic& t) { return t.name; });
+
+        return topics;
+    };
+
+    if (!ctx.audit(std::move(additional_resources_func))) {
+        auto dist = std::distance(begin, valid_range_end);
+        create_topics_response err_resp(
+          error_code::broker_not_available,
+          "Broker not available - audit system failure",
+          std::move(response),
+          std::move(request),
+          dist);
```

Comment:
> EDIT: instead, since dist only applies to request and we are going to return now anyway, could you erase the end of request and then not need to pass in any end point?

Yup good point.